lib/wallet/sync.rs

```diff
@@ -67,28 +67,34 @@ impl WalletInner {
         let start = SystemTime::now();
         tracing::trace!("starting wallet sync");
         // Don't error out here if the wallet is locked, just skip the sync.
-        let wallet_read = match self.read_wallet() {
-            Ok(wallet_read) => wallet_read,
+        let mut wallet_write = match self.write_wallet() {
+            Ok(wallet_write) => wallet_write,
```

Comment:
I don't really understand why this locking behavior is better than the previous one. Would be good with some explanations in comments as to why we're choosing write/read at different steps.